### ISSUE TITLE
Not getting headers in response for android client

### ISSUE THREAD

**Issue by kpradeepkumarreddy** [swagger-api/swagger-codegen#4281]
##### Description
I have generated code for android client that uses volley library.  I want to get headers from response. How can i get the headers from the response ???

##### Swagger-codegen version
swagger-codegen-cli-2.2.1.jar

##### Swagger declaration file content or url
 ```yaml
  paths:
    /login:
     post:
       summary: send mobileNum and otp to login
       description: send mobileNum and otp to login
       parameters:
        - name: args
          in: query
          description: send mobileNum and otp in args 
          required: true
          type: string
          format: string
      tags:
        - login
      responses:
        '200':
          headers:
            Authorization:  
              description: access token will be set in the authorization header
              type: string
          description: login successful, received accessToken in header and refreshToken in body
          schema:
            $ref: '#/definitions/ResponseTO'
      ```

  ##### Command line used for generation
java -jar swagger-codegen-cli.jar generate -i login1.yml -l android --library=volley -o login-source-code/ -c config.json

=> for adding header to the request, i can use the following code,
      LoginApi loginApi = new LoginApi();
      loginApi.addHeader(String key, String value)

=> How can i get the headers from the response ??????

**Comment by wing328** [comment]
@kpradeepkumarreddy there should be a `WithHttpInfo` method (for every endpoint) to get the response headers as well as status code.

**Comment by wing328** [comment]
If you need further help, let us know. Closing the issue for the time being.

**Comment by wing328** [comment]
(If your company is using Swagger Codegen, please consider adding your company name to [this list](https://github.com/swagger-api/swagger-codegen#companiesprojects-using-swagger-codegen))

**Comment by kpradeepkumarreddy** [comment]
Can you please explain your answer little more elaborately with an example.

**Comment by kpradeepkumarreddy** [comment]
Yes my company is using Swagger Codegen. I have added my company name to the list.

**Comment by wing328** [comment]
@kpradeepkumarreddy my bad. We've not yet added that (WithHttpInfo method) to the Android (Volley) API client. Please use the Java API client which comes with the `WithHttpInfo`: https://github.com/swagger-api/swagger-codegen/wiki/FAQ#how-can-i-generate-an-android-sdk

For example:
[addPet](https://github.com/swagger-api/swagger-codegen/blob/master/samples/client/petstore/java/okhttp-gson/src/main/java/io/swagger/client/api/PetApi.java#L111)
[addPetWithHttpInfo](https://github.com/swagger-api/swagger-codegen/blob/master/samples/client/petstore/java/okhttp-gson/src/main/java/io/swagger/client/api/PetApi.java#L122)

**Comment by kpradeepkumarreddy** [comment]
can you please add the WithHttpInfo() method to the android(volley). Because, till now i was using android(volley) and i know how to use the code generated for android(volley).  Now when i generated the code for java (retrofit2 or okhttp-gson), the generated code is different from what is generated for android(volley).  Now with this newly generated code, i don't know how to make a request and get the responses.

using android(volley) generated code i used to make the request and get response as follows:-
--------------------------------------------------------------------------------------------------------

ApiInvoker.initializeInstance();

LoginApi loginApi = new LoginApi();
LoginTO requestArgs = new LoginTO();
requestArgs.setMobileNum(mobileNum);
requestArgs.setOtp((String) params[0]);

  Gson gson = new Gson();
  String reqArgStr = gson.toJson(requestArgs);

ResponseTO responseTO = loginApi.loginPost(reqArgStr);

=> Now the generated code for java(retrofit2 or okhttp-gson) is different. I don't know how to use this code.

**Comment by wing328** [comment]
For Java (okhttp-gson) API client, it comes with documentation and the README.md is a good starting point.

**Comment by wing328** [comment]
> can you please add the WithHttpInfo() method to the android(volley). Because, till now i was using android(volley) and i know how to use the code generated for android(volley). 

May I know if you've time to contribute the enhancement? If yes, I can show you some good starting points.

**Comment by kpradeepkumarreddy** [comment]
yes i have time to contribute the enhancement.

**Comment by wing328** [comment]
@kpradeepkumarreddy thanks for offering help to contribute the enhancements. Here are the starting points.

1) you will need to generate "ApiResponse" model class: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/Java/libraries/okhttp-gson/ApiResponse.mustache

2) you will need to add an "additional" method `{{nickname}}WithHttpInfo`: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/android/libraries/volley/api.mustache#L56

and then the original method (e.g getPetByID) will just call the `WithHttpInfo` method (e.g. getPetByIdWithHttpInfo). Here is what it looks like in Java (okhttp-gson): https://github.com/swagger-api/swagger-codegen/blob/master/samples/client/petstore/java/okhttp-gson/src/main/java/io/swagger/client/api/PetApi.java#L532-L555

Before you start, please pull the latest master.

If you need any help or have any question, please let us know.
